web/src/app/modules/views/side/modules/errors-warnings/components/errors-warnings.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnInit } from '@angular/core';
+import { ValidationResult } from '../../../../../../validation/validation-result';
+import { ValidationService } from '../../../../../forms/modules/validation/services/validation.service';
+import { AdditionalInformationService } from '../../links-actions/services/additional-information.service';
+
+
+@Component({
+    moduleId: module.id.toString(),
+    selector: 'errors-warnings',
+    templateUrl: 'errors-warnings.component.html',
+    styleUrls: ['errors-warnings.component.css']
+})
+export class ErrorsWarings implements OnInit {
+    private _isCollapsed = false;
+    public set isCollapsed(collapsed: boolean) {
+        this._isCollapsed = collapsed;
+    }
+
+    public get isCollapsed() {
+        if (!this.warnings) {
+            return true;
+        }
+        return this._isCollapsed;
+    }
+
+
+    public visible = true;
+    constructor(private validationService: ValidationService,
+                private additionalInformationService: AdditionalInformationService) { }
+
+    ngOnInit() { }
+
+    private _tmpStringSet = new Set();
+    private _tmpList: ValidationResult[][] = [];
+    private get warnings() {
+        if (!this.additionalInformationService.element) {
+            return;
+        }
+        let map = this.validationService.findValidationResults(this.additionalInformationService.element, elem => !elem.isValid);
```

Comment:
You should use more precise names than `map`, `_tmpList` and so on. Right now, the code is not very understandable